ReactAndroid/src/main/java/com/facebook/react/devsupport/MultipartStreamReader.java

```diff
@@ -113,7 +148,10 @@ public boolean readAllParts(ChunkCallback callback) throws IOException {
         Buffer chunk = new Buffer();
         content.skip(chunkStart);
         content.read(chunk, length);
+        emitProgress(currentHeaders, chunk.size() - currentHeadersLength, true, progressCallback);
         emitChunk(chunk, isCloseDelimiter, callback);
+        currentHeaders = null;
```

Comment:
clear these only when we return (or don't clear at all) so we don't always need to reparse?

Reply:
Each chunk of the multipart response has it's own headers so this is why we clear it every time we emit a chunk so we parse the next one.